pkg/cmd/server/server_test.go

```diff
@@ -41,3 +43,51 @@ func TestServerGracefulTerminationOnError(t *testing.T) {
 	_, err = c.Complete(ctx)
 	require.ErrorIs(t, err, context.Canceled)
 }
+
+func TestReplaceMiddleware(t *testing.T) {
+	c := Config{MiddlewareModification: []MiddlewareModification{
+		{
+			Operation: OperationReplaceAllUnsafe,
+			Middlewares: []ReferenceableMiddleware{
+				{
+					Name:                "foobar",
+					UnaryMiddleware:     mockUnaryInterceptor{val: 1}.unaryIntercept,
+					StreamingMiddleware: mockStreamInterceptor{val: errors.New("hi")}.streamIntercept,
+				},
+			},
+		},
+	}}
+	unary, streaming, err := c.buildMiddleware(nil)
+	require.NoError(t, err)
+	require.Len(t, unary, 1)
+	require.Len(t, streaming, 1)
+	val, _ := unary[0](context.Background(), nil, nil, nil)
```

Comment:
I like to put newlines after sets of requires before the next "executing" statement